adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m

```diff
@@ -88,26 +88,35 @@ - (void)renderNativeAdForAdConfiguration:
                  strongSelf->_nativeAd = nativeAd;
                  strongSelf->_nativeAd.delegate = strongSelf;
                  strongSelf->_nativeAd.rootViewController = adConfiguration.topViewController;
-
-                 if (strongSelf->_loadCompletionHandler) {
-                   id<GADMediationNativeAdEventDelegate> delegate =
-                       strongSelf->_loadCompletionHandler(strongSelf, nil);
-                   strongSelf->_delegate = delegate;
-                 }
+      
+                 [strongSelf loadRequireData];
                }];
 }
 
-#pragma mark - GADMediationNativeAd
-
-- (GADNativeAdImage *)icon {
-  if (!_icon) {
-    if (_nativeAd.data.icon && _nativeAd.data.icon.imageURL != nil) {
-      _icon = [self imageWithUrlString:_nativeAd.data.icon.imageURL];
-    }
-  }
-  return _icon;
+- (void)loadRequireData {
+    NSString *urlString = _nativeAd.data.icon.imageURL;
+    dispatch_async(dispatch_get_global_queue(0, 0), ^{
+        NSURL *url = [NSURL URLWithString:urlString];
+        NSData *data = [NSData dataWithContentsOfURL:url];
```

Comment:
You mentioned that the thread was stuck because of this line. Can you clarify about this? Should there be a fix in Pangle SDK if the url cannot be populated?

Reply:
`Don't use this synchronous method to request network-based URLs. For network-based URLs, this method can block the current thread for tens of seconds on a slow network, resulting in a poor user experience, and in iOS, may cause your app to be terminated.` in apple development doc

Reply:
Seems like this is the documentation: https://developer.apple.com/documentation/foundation/nsdata/1547245-datawithcontentsofurl

Then shouldn't we use `[dataTaskWithURL:completionHandler:](https://developer.apple.com/documentation/foundation/nsurlsession/1410330-datataskwithurl)` instead?

Reply:
We use `-dataWithContentsOfURL` asynchronously. Must we use  `[dataTaskWithURL:completionHandler:]`?